AI/scripts/model_main_tf2.py

```diff
@@ -0,0 +1,242 @@
+flags.DEFINE_string(
+    'tpu_name',
+    default=None,
+    help='Name of the Cloud TPU for Cluster Resolvers.')
+flags.DEFINE_integer(
+    'num_workers', 1, 'When num_workers > 1, training uses '
+                      'MultiWorkerMirroredStrategy. When num_workers = 1 it uses '
+                      'MirroredStrategy.')
+flags.DEFINE_integer(
+    'checkpoint_every_n', 50, 'Integer defining how often we checkpoint.')
+flags.DEFINE_boolean('record_summaries', True,
+                     ('Whether or not to record summaries during'
+                      ' training.'))
+
+# NEW FLAGS
+
+flags.DEFINE_string(
+    'name',
+    help='a name that uniquely identifies this mosaic and training data',
+    default='test'
+)  # TODO must be changed to account for user/project
+flags.DEFINE_string(
+    'data_dir',
+    help='directory containing mosaic and CSVs to train on',
+    default=os.path.join(os.path.dirname(__file__), '../test/test')
+)
+flags.DEFINE_string(
+    'model_name',
+    help='the name of the pretrained model from the TF Object Detection model zoo',
+    default='faster_rcnn_resnet50_v1_640x640_coco17_tpu-8'
+)  # TODO maybe add option for year
+flags.DEFINE_boolean(
+    'continue_from_checkpoint',
+    help='whether training should continue from a checkpoint',
+    default=False
+)
+
+FLAGS = flags.FLAGS
+
+
+def main(unused_argv):
+    # TODO what flags should be required?
+    # TODO how to handle checkpoint
+
+    # pretrained model from TensorFlow Object Detection model zoo
+    pretrained_model_dir = os.path.join(ROOT_DIR, 'pre-trained-models', FLAGS.model_name)
+    if not os.path.exists(pretrained_model_dir):
+        download_and_unpack_model(FLAGS.model_name)
+
+    # user models
+    user_models_dir = os.path.join(ROOT_DIR, 'models')
+    create_directory_if_not_exists(user_models_dir)
+
+    # path to directory containing user-trained models for this mosaic
+    mosaic_models_dir = os.path.join(user_models_dir, FLAGS.name)
+    create_directory_if_not_exists(mosaic_models_dir)
+
+    # path to directory containing the specific user-trained model for this mosaic
+    mosaic_model_dir = os.path.join(mosaic_models_dir, FLAGS.model_name)
+    if os.path.exists(mosaic_model_dir):
+        if FLAGS.continue_from_checkpoint:
+            logger.error(f'Cannot find {full_path(mosaic_model_dir)}')
+            sys.exit(1)
+        else:
+            shutil.rmtree(mosaic_model_dir)
+            logger.info(f'Removed {full_path(mosaic_model_dir)}')
+
+    create_directory_if_not_exists(mosaic_model_dir)
+
+    mosaic_annotations_dir = os.path.join(ROOT_DIR, 'annotations/' + FLAGS.name)
+    num_classes = len([f for f in os.listdir(FLAGS.data_dir) if f.lower().endswith('.csv')])
+    pipeline_config_path = edit_pipeline_config(pretrained_model_dir, mosaic_model_dir, num_classes, mosaic_annotations_dir)
+
+
+    # flags.mark_flag_as_required('model_dir')
+    # flags.mark_flag_as_required('pipeline_config_path')
```

Comment:
take this out

Reply:
No, because required flags is not yet settled.

Reply:
Then add a TODO we we know to fix this once we finalize the flag stuff